src/engine/source/rbac/test/src/rbac_test.cpp

```diff
@@ -0,0 +1,189 @@
+#include <gtest/gtest.h>
+
+#include <logging/logging.hpp>
+#include <rbac/rbac.hpp>
+#include <store/mockStore.hpp>
+
+using namespace rbac;
+using namespace store::mocks;
+
+auto constexpr OK_ROLE = "role";
+auto constexpr OK_RESOURCE = Resource::ASSET;
+auto constexpr OK_OPERATION = Operation::READ;
+
+auto constexpr BAD_ROLE = "bad_role";
+auto constexpr BAD_RESOURCE = Resource::SYSTEM_ASSET;
+auto constexpr BAD_OPERATION = Operation::WRITE;
+
+auto MODEL_JSON = json::Json {R"({
+    "role": [
+        {
+            "resource": "asset",
+            "operation": "read"
+        }
+    ],
+    "role2": [
+        {
+            "resource": "asset",
+            "operation": "read"
+        },
+        {
+            "resource": "asset",
+            "operation": "write"
+        }
+    ],
+    "role3": [
+        {
+            "resource": "system_asset",
+            "operation": "read"
+        },
+        {
+            "resource": "system_asset",
+            "operation": "write"
+        },
+        {
+            "resource": "asset",
+            "operation": "read"
+        },
+        {
+            "resource": "asset",
+            "operation": "write"
+        }
+    ]
+})"};
+auto MODEL_ERROR_JSON = json::Json {"{}"};
+
+/******************************************************************************/
+/* General tests */
+/******************************************************************************/
+class RBACTest : public ::testing::Test
```

Comment:
improve - desing: apply TEST_P

Reply:
We do not believe that it is the best strategy to test the load and shadow tests, the paths within the code are different so the effort would not really improve the readability of the test.
On the other hand, the RBAC module is pending implementation, so it will continue to be improved when it is used.